tangos/live_calculation/builtin_functions/reassembly.py

```diff
@@ -1,23 +1,24 @@
 from __future__ import absolute_import
 from tangos.core import extraction_patterns
 from . import BuiltinFunction
-from .. import StoredProperty, FixedInput
+from .. import StoredProperty, FixedInput, LiveProperty
 
 
 @BuiltinFunction.register
 def raw(halos, values):
     return values
-raw.set_input_options(0, assert_class=StoredProperty)
 
 @raw.set_initialisation
 def raw_initialisation(input):
-    input.set_extraction_pattern(extraction_patterns.halo_property_raw_value_getter)
+    if isinstance(input, LiveProperty):
```

Comment:
You want to avoid things like this wherever possible as it spreads around knowledge of the inner workings of a `LiveProperty` as opposed to a normal value retrieval. That knowledge belongs in the `LiveProperty` class itself.

I suggest always calling `set_raw`. Create a `set_raw` method for the base `Calculation` object, which is then overriden in `LiveProperty` and also overriden in `StoredProperty`. The `StoredProperty` implementation just wants to call `set_extraction_pattern(...)`

Reply:
I don't think this issue was resolved